app/helpers/template_helpers.py

```diff
@@ -1,304 +1,378 @@
-    }
+            "google_tag_manager_id": self._google_tag_manager_id,
+            "google_tag_manager_auth": self._google_tag_manager_auth,
+        }
+
+    @property
+    def page_header_context(self) -> dict[str, Any]:
+        context = {
+            "logo": f"{self.context_options.page_header_logo}",
+            "logoAlt": f"{self.context_options.page_header_logo_alt}",
+        }
 
-    if request.blueprint == "post_submission":
-        default_context["footerWarning"] = lazy_gettext(
-            "Make sure you <a href='{sign_out_url}'>leave this page</a> or close your browser if using a shared device"
-        ).format(sign_out_url=sign_out_url)
-
-    context = {
-        "default": {
-            **default_context,
-        },
-        "social": {
-            **default_context,
-        },
-        "northernireland": {
-            **default_context,
-        },
-        "census": {
-            **default_context,
-        },
-        "census-nisra": {
-            **default_context,
-            "lang": "en",
-            "poweredBy": {
-                "logo": "nisra-logo-black-en",
-                "alt": "NISRA - Northern Ireland Statistics and Research Agency",
+        if self.context_options.title_logo:
+            context["titleLogo"] = f"{self.context_options.title_logo}"
+        if self.context_options.title_logo_alt:
+            context["titleLogoAlt"] = f"{self.context_options.title_logo_alt}"
+        if self.context_options.header_logo:
+            context["customHeaderLogo"] = self.context_options.header_logo
+        if self.context_options.mobile_logo:
+            context["mobileLogo"] = self.context_options.mobile_logo
+
+        return context
+
+    @property
+    def footer_context(self):
+        context = {
+            "lang": self._language,
+            "crest": self.context_options.crest,
+            "newTabWarning": lazy_gettext("The following links open in a new tab"),
+            "copyrightDeclaration": {
+                "copyright": self.context_options.copyright_declaration,
+                "text": self.context_options.copyright_text,
             },
-        },
-    }
-    return context.get(theme)
+        }
 
+        if self._footer_warning:
+            context["footerWarning"] = self._footer_warning
+
+        if self.context_options.footer_links:
+            context["rows"] = [{"itemsList": self.context_options.footer_links}]
+
+        if self.context_options.footer_legal_links:
+            context["legal"] = [{"itemsList": self.context_options.footer_legal_links}]
+
+        if (
+            self.context_options.powered_by_logo
+            or self.context_options.powered_by_logo_alt
+        ):
+            context["poweredBy"] = {
+                "logo": self.context_options.powered_by_logo,
+                "alt": self.context_options.powered_by_logo_alt,
+            }
+
+        return context
+
+    @cached_property
+    def _footer_warning(self) -> Union[str, None]:
```

Comment:
Style guide was to only use Union with None, when it is more than 1 type excluding None.
```suggestion
    def _footer_warning(self) -> Optional[str]:
```